tracer/src/Datadog.Tracer.Native/debugger_method_rewriter.cpp

```diff
@@ -667,27 +681,38 @@ HRESULT DebuggerMethodRewriter::ApplyMethodProbe(
         rewriterWrapper.LoadLocal(returnValueIndex);
     }
 
-    // Changes all returns to a LEAVE.S
+    // Changes all returns to a LEAVE.S (including branches to `ret`)
     for (ILInstr* pInstr = rewriter.GetILList()->m_pNext; pInstr != rewriter.GetILList(); pInstr = pInstr->m_pNext)
     {
-        switch (pInstr->m_opcode)
+        if (pInstr->m_opcode == CEE_RET && pInstr != methodReturnInstr)
+        {
+            if (!isVoid)
+            {
+                rewriterWrapper.SetILPosition(pInstr);
+                rewriterWrapper.StLocal(returnValueIndex);
+            }
+            pInstr->m_opcode = CEE_LEAVE_S;
+            pInstr->m_pTarget = endFinallyInstr->m_pNext;
+        }
+        else if (ILRewriter::IsBranchTarget(pInstr) && pInstr->m_pTarget->m_opcode == CEE_RET)
         {
-            case CEE_RET:
+            if (!isVoid)
             {
-                if (pInstr != methodReturnInstr)
+                rewriterWrapper.SetILPosition(pInstr);
+                rewriterWrapper.StLocal(returnValueIndex);
+
+                // unconditional branching instructions (`br`) are not popping any value from the top of the stack.
+                // other conditional branches, though, are mutating the evaluation stack (e.g `brtrue` pops
+                // the top of the stack and jumps to the target if it's non-zero/true).
+                // If we are dealing with conditional branches that changes the evaluation stack, we fix
+                // the evaluation stack accordingly by loading it back in case we are not branching.
+                if (pInstr->m_opcode != CEE_BR && pInstr->m_opcode != CEE_BR_S &&
```

Comment:
```suggestion
                // Unconditional branching instructions (`br`) do not pop any value from the top of the stack.
                // Other conditional branches, though, do mutate the evaluation stack (e.g `brtrue` pops
                // the top of the stack and jumps to the target if it's non-zero/true).
                // If we are dealing with conditional branches that changes the evaluation stack, we fix
                // the evaluation stack accordingly by loading it back in case we are not branching.
                if (pInstr->m_opcode != CEE_BR && pInstr->m_opcode != CEE_BR_S &&
```

Reply:
Done in bf6df15958c13137a9c5bbf36c849990bc3d909e.